core/engine/src/bytecompiler/class.rs

```diff
@@ -423,21 +423,33 @@ impl ByteCompiler<'_> {
                             self.emit_with_varying_operand(Opcode::PushClassPrivateSetter, index);
                         }
                         MethodDefinition::Ordinary(expr) => {
+                            self.emit(Opcode::RotateLeft, &[Operand::U8(3)]);
+                            self.emit_opcode(Opcode::Dup);
+                            self.emit(Opcode::RotateRight, &[Operand::U8(4)]);
```

Comment:
Thought: I have seen this pattern in a lot of places. Maybe it would be good to extract this operation into a common compiler method. Something like `dup_indexed` or similar.

Reply:
Totaly agree. I think this might be something for the bytecode optimizer too. I would wait for https://github.com/boa-dev/boa/pull/3798 until we work on that since that is probably gonna change some things.